### DISCUSSION TITLE
How to generate a data table？

### DISCUSSION THREAD

**Discussion by kenanaptx4869** [mss-boot-io/mss-boot-admin#99] [Q&A] [answered]
How to generate a data table？ or .sql file ?

**Comment by lwnmengjing** [comment]
If there is no GO language development ability, you can use the virtual model to configure the model, and then generate data; if you can develop the Go language, you can directly write a model, add it to Migrate, run the command command, run the command command`go run . migrate`

**Comment by lwnmengjing** [comment, marked as answer]
Update in 2026: the virtual model path has been downgraded and is on the deprecation path, so it should not be the main recommendation for new usage anymore.

For new projects, please prefer the Go model + migration workflow:

```bash
go run . migrate
```

Use the virtual model only for existing installations that already depend on it. If you are building a new data table now, defining the model in Go keeps the schema, migration, tests, and agent-readable code path easier to maintain.
